pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1168,6 +1168,7 @@ public void updateKafkaPartitionsIfNecessary(String realtimeTableName, TableConf
     // Generate new kafka partition assignment and update the znode
     if (currentInstances.size() < currentReplicaCount) {
       LOGGER.error("Cannot have {} replicas in {} instances for {}.Not updating partition assignment", currentReplicaCount, currentInstances.size(), realtimeTableName);
+      _controllerMetrics.addMeteredTableValue(realtimeTableName, ControllerMeter.SHORT_OF_LIVE_INSTANCES, 1);
```

Comment:
You should add an else clause in which you set the gauge back to 0. Otherwise the alert will trigger forever